test/extended/openstack/servergroup.go

```diff
@@ -113,3 +155,28 @@ func serverGroupIDsFromName(client *gophercloud.ServiceClient, name string) ([]s
 
 	return IDs, nil
 }
+
+type installConfig struct {
+	ControlPlane struct {
+		Platform struct {
+			OpenStack struct {
+				ServerGroupPolicy string `yaml:"serverGroupPolicy"`
+			} `yaml:"openstack"`
+		} `yaml:"platform"`
+	} `yaml:"controlPlane"`
+}
+
+func installConfigFromCluster(client clientcorev1.ConfigMapsGetter) (installConfig, error) {
+	const installConfigName = "cluster-config-v1"
+
+	cm, err := client.ConfigMaps("kube-system").Get(context.Background(), installConfigName, metav1.GetOptions{})
+	if err != nil {
+		return installConfig{}, err
+	}
+
+	var ic installConfig
+	err = yaml.Unmarshal([]byte(cm.Data["install-config"]), &ic)
+	return ic, err
+
+	return ic, nil
```

Comment:
This line is not reachable; you are already returning nil on line 179 if Unmarshal was successful.

Reply:
You can safely delete this line (and line 180 just above).